staging/src/k8s.io/client-go/features/known_features.go

```diff
@@ -68,6 +81,11 @@ func LibraryFeatureSet() featuregates.FeatureSet {
 	return libraryFeatureSet
 }
 
+// FIXME: rename to FeatureGates when other definition is gone.
+func LibFeatureGates() *feature.GateSet {
```

Comment:
I see an advantage to using an interface over a concrete type here so that in composition people can say, "David and Tim won't are too slow to merge my awesome way to set FeatureGates, so I'll provide a compatible implementation myself!" and anything accepting a GateSet will "just work".

Reply:
I see that as exceedingly unlikely, or at least not super compelling vs. the complexity.  Using interfaces has a few costs, IMO:

1) Harder to keep in brain
2) MUCH harder to back-track to "where did I get called from"
3) Requires everything be methods, meaning the types get more obtuse
4) Encourages defining more and more interfaces when a "dumb" struct would do
5) Encourages "SimpleFeatureGate" stuff which makes it more difficult again to track

Aside: I looked again at your branch and I honestly have no idea what a private method in a public interface means :)

I won't die on this hill, probably, but this seems like the sort of API where simpler really is better.

I'll try to take a pass over this variation and see if there's a not-so-bad way to have the best of both